commit 1ae978208e2ee9ba1b01d309164bc5e590cd242d
Author: Herbert Xu <herbert@gondor.apana.org.au>
Date:   Thu Aug 30 15:36:14 2007 +0800

    [CRYPTO] api: Add aead crypto type
    
    This patch adds crypto_aead which is the interface for AEAD
    (Authenticated Encryption with Associated Data) algorithms.
    
    AEAD algorithms perform authentication and encryption in one
    step.  Traditionally users (such as IPsec) would use two
    different crypto algorithms to perform these.  With AEAD
    this comes down to one algorithm and one operation.
    
    Of course if traditional algorithms were used we'd still
    be doing two operations underneath.  However, real AEAD
    algorithms may allow the underlying operations to be
    optimised as well.
    
    Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>

diff --git a/crypto/Makefile b/crypto/Makefile
index a070dcc074ff..9821c5ba054e 100644
--- a/crypto/Makefile
+++ b/crypto/Makefile
@@ -9,6 +9,7 @@ crypto_algapi-objs := algapi.o $(crypto_algapi-y)
 obj-$(CONFIG_CRYPTO_ALGAPI) += crypto_algapi.o
 
 obj-$(CONFIG_CRYPTO_ABLKCIPHER) += ablkcipher.o
+obj-$(CONFIG_CRYPTO_AEAD) += aead.o
 obj-$(CONFIG_CRYPTO_BLKCIPHER) += blkcipher.o
 
 crypto_hash-objs := hash.o